packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -265,6 +266,35 @@ export function isTaggedTelemetryPropertyValue(x: any): x is ITaggedTelemetryPro
     return (typeof (x?.value) !== "object" && typeof (x?.tag) === "string");
 }
 
+/**
+ * Filter loggable telemetry properties
+ * @param x - any telemetry prop
+ * @returns - as-is if x is primitive. returns stringified if x is an array of primitive.
+ * otherwise returns null since this is what we support at the moment.
+ */
+// eslint-disable-next-line @rushstack/no-new-null
+function filterValidTelemetryProps(x: any): TelemetryEventPropertyType | null {
+    if (Array.isArray(x) && x.every((val) => isPrimitive(val))) {
+        return JSON.stringify(x);
+    }
+    if (isPrimitive(x)) {
+        return x;
+    }
+    return null;
+}
+
+// checking type of x, returns false if x is null
+function isPrimitive(x: any): x is TelemetryEventPropertyType {
+    switch (typeof x) {
+        case "string":
+        case "number":
+        case "boolean":
+        case "undefined":
+            return true;
+        default:
+            return false;
```

Comment:
Logically, default case should recursively call isPrimitive() for each of the elements of an array, if that's an array. And name of the function should be something like "isTelemetrySerializable"

filterValidTelemetryProps would not need to check elements of array that way, but it might still want (as it does right now) to use different serialization strategies for arrays vs. non-array types.

Reply:
i think we don't want recursion and we don't want to allow nested arrays. per Tony's comment above.
I can definitely rename it